File: q1351/q1351.cpp
class Solution
{
public:
    int countNegatives(vector<vector<int>> &grid)
    {
        int res = 0;
        int first_negative_index = grid[0].size() - 1;
        for (auto &row : grid)
        {
            // find the first time a negative element appear in the row
            while (first_negative_index >= 0 && row[first_negative_index] < 0)
            {
                // the next row's first negative element index will be smaller
                first_negative_index--;
            }
            // update the result
            res += (row.size() - (first_negative_index + 1));
        }
        return res;
    }
};

File: q1351/q1351.java
class Solution {
    public int countNegatives(int[][] grid) {
        int firstNegativeIdx = grid[0].length - 1;
        int totalNegatives = 0;
        for (int[] row : grid) {
            // find the first negative index from this row
            while (firstNegativeIdx >= 0 && row[firstNegativeIdx] < 0) {
                firstNegativeIdx--;
            }
            // compute the number of negative numbers in this row
            totalNegatives += row.length - firstNegativeIdx - 1;
        }
        return totalNegatives;
    }
}